lib/logflare/backends/recent_events_touch.ex

```diff
@@ -0,0 +1,99 @@
+defmodule Logflare.Backends.RecentEventsTouch do
+  @moduledoc """
+  Periodically updates a Source.log_events_updated_at field to the latest ingested_at timestamp.
+
+  Source.log_events_updated_at is used for determining sources to warm on node startup.
+  """
+  use TypedStruct
+  use GenServer
+
+  alias Logflare.Sources
+  alias Logflare.Backends
+  alias Logflare.Sources
+
+  require Logger
+
+  ## Server
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args,
+      name: Backends.via_source(args[:source], __MODULE__),
+      hibernate_after: 5_000,
+      spawn_opt: [
+        fullsweep_after: 100
+      ]
+    )
+  end
+
+  ## Client
+  def init(args) do
+    source = Keyword.get(args, :source)
+
+    Process.flag(:trap_exit, true)
+    Logger.metadata(source_id: source.token, source_token: source.token)
+    touch_every = args[:touch_every]
+    touch(touch_every)
+
+    Logger.debug("[#{__MODULE__}] Started")
+
+    {:ok,
+     %{
+       source_token: source.token,
+       source_id: source.id,
+       touch_every: touch_every
+     }}
+  end
+
+  defp random_interval_ms() do
+    min = :timer.minutes(1)
+    max = :timer.minutes(55)
+    Enum.random(min..max)
+  end
+
+  def handle_info(:touch, %{source_id: source_id} = state) do
+    source =
+      source_id
+      |> Sources.Cache.get_by_id()
+
+    Backends.list_recent_logs_local(source)
```

Comment:
btw I think this is one thing that kills us.

Can we just do this logic with an ETS query instead?

Because when we return all the log events from ETS it is copying these large JSONs every time.

Reply:
Also, I am on prod and this fn is giving me 200 log events when it seems like it should only give me 100.

```
source_token |> Logflare.Sources.get_source_by_token() |> Logflare.Backends.list_recent_logs_local() |> Enum.count()
200
```

Reply:
> this fn is giving me 200 log events when it seems like it should only give me 100.
This function takes 100 events from each queue, so if there are two broadway pipelines there would be 200 events.
We could do the sort+take within the func but decided to let the caller handle it instead...


Yea placing it in a Task is fine in this case

Reply:
`Backends.list_recent_logs_local(source)` is actually multiple sequential ets queries, have benchmarked it and it isn't slow. The variable binding would have some minor memory impact ofc, definitely can fire off a task for this.  Would be able to tone down the GCing if so.

Reply:
fixed both issues.